test/gtest/core/test_alltoall.cc

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include "common/test_ucc.h"
+
+// tmp
+#ifdef ucc_max
+#undef ucc_max
+#endif
+
+#include "src/utils/ucc_math.h"
+
+using Param_0 = std::tuple<int, int, int>;
+using Param_1 = std::tuple<int, int>;
+
+class test_alltoall : public UccCollArgs, public ucc::test
+{
+public:
+    UccCollArgsVec data_init(int nprocs, ucc_datatype_t dtype,
+                                     size_t count) {
+        UccCollArgsVec args;
```

Comment:
minor, you can allocate vector of size nprocs instead of increasing the size each time in push_back